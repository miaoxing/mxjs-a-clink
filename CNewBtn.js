import {NewBtn} from '@mxjs/a-button';
import curUrl from '@mxjs/cur-url';

const getPermission = () => curUrl.new();

const CNewBtn = (props) => (
  <NewBtn href={getPermission()} {...props}/>
);

CNewBtn.getPermission = getPermission;

export default CNewBtn;
