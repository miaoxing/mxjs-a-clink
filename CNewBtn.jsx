import {NewBtn} from '@mxjs/a-button';
import curUrl from '@mxjs/cur-url';

const getPermission = () => curUrl.new();

const CNewBtn = (props) => (
  <NewBtn to={getPermission()} {...props}/>
);

CNewBtn.getPermission = getPermission;

export default CNewBtn;
