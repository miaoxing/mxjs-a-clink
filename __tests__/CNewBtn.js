import React from 'react';
import CNewBtn from '../CNewBtn';
import {MemoryRouter} from 'react-router';
import {render} from '@testing-library/react';

describe('CNewBtn', () => {
  test('basic', () => {
    const result = render(<MemoryRouter>
      <CNewBtn/>
    </MemoryRouter>);

    expect(result.container).toMatchSnapshot();
  });
});
