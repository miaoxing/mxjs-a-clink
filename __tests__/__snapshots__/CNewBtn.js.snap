// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CNewBtn basic 1`] = `
<div>
  <a
    class="btn btn-success"
    href="/"
  >
    添 加
  </a>
</div>
`;
